pkg/yamlgen/yamlgen.go

```diff
@@ -0,0 +1,243 @@
+		jen.Qual("structgen/cfggen", "Generate").Call(jen.Id("config"), jen.Qual("os", "Stderr")),
+	))
+
+	// Generate main function in new module.
+	generatedCode.Func().Id("main").Params().Block(
+		init...,
+	)
+	return fmt.Sprintf("%#v", generatedCode), nil
+}
+
+// execGoCode executes and returns output from generated Go code.
+func ExecGoCode(ctx context.Context, mainGo string) ([]byte, error) {
+	tmpDir, err := ioutil.TempDir("", "structgen")
+	if err != nil {
+		return nil, err
+	}
+	defer os.RemoveAll(tmpDir)
+
+	// TODO(saswatamcode): Remove once merged.
+	// This is weird but need it for getting Generate function inside tmpDir in PR.
+	// Once merged this can be removed and can be replaced with just an import in tmpDir/main.go.
+	err = os.Mkdir(filepath.Join(tmpDir, "cfggen"), 0700)
+	if err != nil {
+		return nil, err
+	}
+	code, err := os.Create(filepath.Join(tmpDir, "cfggen/cfggen.go"))
+	if err != nil {
+		return nil, err
+	}
+	defer code.Close()
+	_, err = code.Write([]byte(cfggenFile))
+	if err != nil {
+		return nil, err
+	}
+
+	// Copy generated code to main.go.
+	main, err := os.Create(filepath.Join(tmpDir, "main.go"))
+	if err != nil {
+		return nil, err
+	}
+	defer main.Close()
+
+	_, err = main.Write([]byte(mainGo))
+	if err != nil {
+		return nil, err
+	}
+
+	// Create go.mod in temp dir.
+	cmd := exec.CommandContext(ctx, "go", "mod", "init", "structgen")
+	cmd.Dir = tmpDir
+	if err := cmd.Run(); err != nil {
+		return nil, errors.Wrapf(err, "run %v", cmd)
+	}
+
+	// Import required packages(generate go.sum).
+	cmd = exec.CommandContext(ctx, "go", "mod", "tidy")
+	cmd.Dir = tmpDir
+	if err := cmd.Run(); err != nil {
+		return nil, errors.Wrapf(err, "run %v", cmd)
+	}
+
+	// Execute generate code and return output.
+	b := bytes.Buffer{}
+	cmd = exec.CommandContext(ctx, "go", "run", "main.go")
+	cmd.Dir = tmpDir
+	cmd.Stderr = &b
+	cmd.Stdout = &b
+	if err := cmd.Run(); err != nil {
+		return nil, errors.Wrapf(err, "run %v out %v", cmd, b.String())
+	}
+
+	return b.Bytes(), nil
+}
+
+// TODO(saswatamcode): Remove once merged.
+// This is weird but need it for getting Generate function inside tmpDir in PR.
+// Could also do with commit hash and go.mod, but it would change on each commit in PR).
+// Once merged this can be removed and can be replaced with just an import in tmpDir/main.go.
+const cfggenFile = `package cfggen
+
+import (
+	"io"
+	"reflect"
+
+	"github.com/fatih/structtag"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+)
+
+func Generate(obj interface{}, w io.Writer) error {
+	// We forbid omitempty option. This is for simplification for doc generation.
+	if err := checkForOmitEmptyTagOption(obj); err != nil {
+		return errors.Wrap(err, "invalid type")
+	}
+	return yaml.NewEncoder(w).Encode(obj)
+}
+
+func checkForOmitEmptyTagOption(obj interface{}) error {
+	return checkForOmitEmptyTagOptionRec(reflect.ValueOf(obj))
+}
+
+func checkForOmitEmptyTagOptionRec(v reflect.Value) error {
```

Comment:
Do we need to check for emit empty if we fill all the structs? I think in this case we need to also set non default values for other things that slices... so maybe for next iteration